tests/unit/MigrateTest.php

```diff
@@ -34,6 +35,19 @@ protected function _after()
 	 * @throws Exception
 	 */
 	public function testMigrate() {
+		$this->assertInstanceOf(MysqlConnection::class, $this->app->connection);
+		$this->assertTrue(method_exists($this->app->connection, 'getPdo'));
+		$this->assertInstanceOf(\PDO::class, $this->app->connection->pdo);
+		$migrationTable = $this->app->connection->migrationTable ?: 'migration';
```

Comment:
Probably that we should have two test paths: one where `$this->app->connection->migrationTable` is set accordingly and the other where it is not (so that `'migration'` is used).

Reply:
It would be too much work to test such a subtle differences. If covers the most important cases, the existing of the migration table and the empty database (after reset), Later it can be completed with some critical cases if something comes up. Complete coverage is not even recommended, because consumes more resource than it resolves. [https://codeception.com/docs/Codecoverage](https://codeception.com/docs/Codecoverage)

Reply:
More than code coverage, this has to do with API contracts and (non-)regression testing. But I understand, it seems to me that the deeper the unit tests are semantically, the more complex they get (depending on what we want to test)…